simulation/halsim_ws_server/src/test/native/cpp/WebServerClientConnectionTest.cpp

```diff
@@ -0,0 +1,107 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "WebServerClientConnectionTest.h"
```

Comment:
This could get consolidated into the `WebServerClientTest` object

Reply:
Did a first pass at refactoring to one class. Please review again.

Reply:
I think it's looking good now. I will give on last pass later today to see if I can keep the web server and client up throughout all tests. But committing for now to have a stable version.

Reply:
I can refactor to have the webserver and client up in main. But to do so I'd have to add my own static var for the webserver because in HALSimWeb::Main after executing MainLoop it calls SetInstance(nullptr). I'm not sure of the reasoning here. We might want to revisit and move this to HALSimWeb::Exit instead. If that is done, it would make the test code much cleaner but I'm unsure of the side effects created for production execution.

Reply:
Making the changes above work and allow for refactoring but it doesn't really save much. The loop needs to be restarted and stopped each test and therefore so does the web socket connection. I don't think we gain much to move it. If you are good @zhiquanyeo , we can resolve this conversation.